runtime/lua/vim/lsp/log.lua

```diff
@@ -117,7 +126,7 @@ end
 ---@param level number log level
 ---@returns (bool) true if would log, false if not
 function log.should_log(level)
-  return level >= current_log_level
+  return current_log_level ~= -1 and level >= current_log_level
```

Comment:
to simplify the logic everywhere, make OFF a really big number. then we don't need the special cases everywhere